lib/diff/lcs/internals.rb

```diff
@@ -60,22 +59,18 @@ def lcs(a, b)
     end
 
     # Now, compute the equivalence classes of positions of elements.
+    # An explanation for how this works: https://codeforces.com/topic/92191/en7
     b_matches = position_hash(b, b_start..b_finish)
 
     thresh = []
     links  = []
-    string = a.kind_of?(String)
 
     (a_start..a_finish).each do |i|
-      ai = string ? a[i, 1] : a[i]
+      ai = a[i]
```

Comment:
We can access string element using its index, so the check doesn't seem necessary

Reply:
Which one do you suggest in this case? I'd be happy to follow your advice on this one.

Reply:
Reverting this specific change is the _easiest_ thing to do right now. Once we have this sorted, we can look at splitting this for performance reasons.

Reply:
I reverted it :)